### ISSUE TITLE
Float tags in BAM files interpreted incorrectly

### ISSUE THREAD

**Issue by Ally-Vella** [igvteam/igv.js#1901]
Dear Developers,

Hello, and thank you for your amazing work on the IGV.js! We encountered a potential issue while trying to add a DIY feature, which requires to read tags from BAM files. There is a discrepancy between the actual values in the BAM file and the values retrieved using alignment.tags()[tag].

Here’s a snippet from the BAM file for reference:

```
92062f62-40a3-42ca-89ca-c79168f4ab5f    16      chr11   2159771 60      63S5M2D52M1D4M1D1M2D7M1D1M2D4M1D2M2I24M1D6M7D18M3D17M1D22M1I17M1D24M787N37M1I17M1D7M1D2M1D138M179N13M1D21M107S ... pa:f:228.71
e0d2b8c4-0819-4be0-92ff-54c21b5d33eb    16      chr11   2159775 60      64S14M1D29M1I8M2D19M2D13M1D1M1D28M1D38M2D35M2D8M1I13M1I5M787N13M1I2M1I32M5I3M1D65M2D1M2D13M2D16M1D12M1I7M1D19M117S ... pa:f:208.37
```

However, the values we retrieved for alignment.tags()['pa'] were -362.7832946777344 and -0.00005300367411109619 (I am not sure about the order). I suspect this might be related to encoding, though I’m not very familiar with the technical details. While troubleshooting, I found the readFloat() function and sought help from ChatGPT, which suggested the following modification:

```js
function readFloat(ba, offset) {
    const dataView = new DataView(ba.buffer, ba.byteOffset, ba.byteLength)
    return dataView.getFloat32(offset, true)
}
```

After applying this change, the issue seemed to be resolved. However, I’m unsure if this was specific to my configuration (as I’m not very experienced in this area), but I thought it might be worth reporting. (And the title is adapted from a previous issue addressing a similar problem \_(:з」∠)_

Thank you again for developing such a fantastic tool!

Best regards,
Ally

**Comment by jrobinso** [comment]
Thanks for the report and the debugging.   I think you are correct, it is the "true" argument to getFloat32 that is key.  This specifies the "ENDIANESS".    However it won't always be true (little endian), although it usually is.   To do this really correctly we need to know the endianess of the BAM file,  which is known but not at this level.   So in short the proposed fix is usually the correct one, but some refactoring here is in order.   Leave this issue open until I have time to do that.

**Comment by jrobinso** [comment]
Oh never mind, all multi-byte numbers in a BAM file are little endian.   So your solution is correct in all cases.

**Comment by Ally-Vella** [comment]
Thanks for your quick responses! I'm really glad I can helped out.

**Comment by jrobinso** [comment]
This fix is released in version 3.0.8
